src/lib.rs

```diff
@@ -78,6 +78,59 @@ impl PartialEq for general::__kernel_timespec {
 #[cfg(feature = "general")]
 impl Eq for general::__kernel_timespec {}
 
+mod cmsg_macros {
+    use crate::ctypes::{c_long, c_uint, c_uchar};
+    use crate::general::{cmsghdr, msghdr};
+    use core::mem::size_of;
+    use core::ptr;
+
+    pub unsafe fn CMSG_ALIGN(len: c_uint) -> c_uint {
+        let c_long_size = size_of::<c_long>() as c_uint;
+        (len + c_long_size - 1) & !(c_long_size - 1)
+    }
+
+    pub unsafe fn CMSG_DATA(cmsg: *const cmsghdr) -> *mut c_uchar {
+        (cmsg as *mut c_uchar).offset(size_of::<cmsghdr>() as isize)
+    }
+
+    pub unsafe fn CMSG_SPACE(len: c_uint) -> c_uint {
+        size_of::<cmsghdr>() as c_uint + CMSG_ALIGN(len)
+    }
+
+    pub unsafe fn CMSG_LEN(len: c_uint) -> c_uint {
+        size_of::<cmsghdr>() as c_uint + len
+    }
+
+    pub unsafe fn CMSG_FIRSTHDR(mhdr: *const msghdr) -> *mut cmsghdr {
+        if (*mhdr).msg_controllen < size_of::<cmsghdr>() as _ {
+            return ptr::null_mut();
+        }
+
+        (*mhdr).msg_control as *mut cmsghdr
+    }
+
+    pub unsafe fn CMSG_NXTHDR(mhdr: *const msghdr, cmsg: *const cmsghdr) -> *mut cmsghdr {
+        // TODO: This function should observe Rust's provenance rules once they're set in stone.
```

Comment:
Could you mention that the only thing this code needs to do is cast pointers to `usize` for doing bounds checks?